File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    private static final int MOD = 1234567;
    public int solution(int n) {
        if (n == 1) return 1;
        if (n == 2) return 1;

        // DP 배열 초기화
        int[] dp = new int[n + 1];
        dp[1] = 1;
        dp[2] = 1;

        // DP를 사용하여 피보나치 수 계산
        for (int i = 3; i <= n; i++) {
            dp[i] = (dp[i - 1] + dp[i - 2]) % MOD;
        }

        return dp[n];
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    MOD = 1234567
    
    # n이 1 또는 2일 때
    if n == 1 or n == 2:
        return 1
    
    # DP 배열 초기화
    dp = [0] * (n + 1)
    dp[1] = 1
    dp[2] = 1
    
    # DP를 사용하여 피보나치 수 계산
    for i in range(3, n + 1):
        dp[i] = (dp[i - 1] + dp[i - 2]) % MOD
    
    return dp[n]